src/components/Popover/Popover.tsx

```diff
@@ -1,249 +1,130 @@
-    {
-        initialOpen = false,
-        disabled = false,
-        autoclosable = true,
-        openOnHover = true,
-        delayOpening,
-        delayClosing,
-        behavior = PopoverBehavior.Delayed,
-        placement,
-        offset = {},
-        tooltipOffset,
-        tooltipClassName,
-        theme = 'info',
-        size = 's',
-        hasArrow = true,
-        hasClose = false,
-        className,
-        children,
-        title,
-        content,
-        htmlContent,
-        contentClassName,
-        links,
-        forceLinksAppearance = false,
-        tooltipActionButton,
-        tooltipCancelButton,
-        onOpenChange,
-        onCloseClick,
-        onClick,
-        anchorRef,
-        anchorElement,
-        strategy,
-        qa,
-        disablePortal = false,
-        tooltipId,
-        focusTrap,
-        autoFocus,
-        restoreFocusRef,
-    },
-    ref,
-) {
-    const direction = useDirection();
-    const controlRef = React.useRef<HTMLDivElement>(null);
-    const closedManually = React.useRef(false);
-    const shouldBeOpen = React.useRef(initialOpen);
-
-    const {
-        isOpen,
-        closingTimeout,
-        openTooltip,
-        openTooltipDelayed,
-        unsetOpeningTimeout,
-        closeTooltip,
-        closeTooltipDelayed,
-        unsetClosingTimeout,
-    } = useOpen({
-        initialOpen,
-        disabled,
-        autoclosable,
-        onOpenChange,
-        delayOpening,
-        delayClosing,
-        behavior,
-        shouldBeOpen,
-    });
-
-    const popupPlacement = React.useMemo<PopupPlacement>(() => {
-        if (placement) {
-            return placement;
-        }
-
-        return direction === 'rtl' ? ['left', 'bottom'] : ['right', 'bottom'];
-    }, [direction, placement]);
-
-    React.useImperativeHandle(
-        ref,
-        () => ({
-            openTooltip,
-            closeTooltip,
-        }),
-        [openTooltip, closeTooltip],
+import type {PopupProps} from '../Popup';
+import type {AriaLabelingProps, DOMProps, QAProps} from '../types';
+import {block} from '../utils/cn';
+import {getElementRef} from '../utils/getElementRef';
+
+export interface PopoverProps
+    extends AriaLabelingProps,
+        QAProps,
+        DOMProps,
+        Pick<
+            PopupProps,
+            | 'strategy'
+            | 'placement'
+            | 'offset'
+            | 'keepMounted'
+            | 'hasArrow'
+            | 'initialFocus'
+            | 'returnFocus'
+            | 'disableFocusVisuallyHiddenDismiss'
+        > {
+    children:
+        | ((props: Record<string, unknown>, ref: React.Ref<HTMLElement>) => React.ReactElement)
+        | React.ReactElement;
+    open?: boolean;
+    onOpenChange?: (open: boolean) => void;
+    disabled?: boolean;
+    content?: React.ReactNode;
+    trigger?: 'click';
+    openDelay?: number;
+    closeDelay?: number;
+    enableSafePolygon?: boolean;
```

Comment:
Could you provide description for this option?

Reply:
There is a note about it in Properties table in readme